tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/EmailSend.scala

```diff
@@ -267,4 +270,11 @@ case class EmailSendResults(created: Option[Map[EmailSendCreationId, EmailSendCr
 case class EmailSendResponse(accountId: AccountId,
                              newState: UuidState,
                              created: Option[Map[EmailSendCreationId, EmailSendCreationResponse]],
-                             notCreated: Option[Map[EmailSendCreationId, SetError]])
\ No newline at end of file
+                             notCreated: Option[Map[EmailSendCreationId, SetError]])
+
+case class MimeMessageSourceImpl(name: String, message: Message) extends MimeMessageSource {
+
+  override def getSourceId: String = name
+
+  override def getInputStream: InputStream = new ByteArrayInputStream(DefaultMessageWriter.asBytes(message))
```

Comment:
Here we write the message as bytes twice:

 - One time here to send it
 - One time when we append it to the mailbox
 
 Maybe we could be smarter and output it as bytes only once?